csrc/scheduler/mma_utils.cpp

```diff
@@ -1300,18 +1326,27 @@ RolesMapOpt getTensorsRoles(Fusion* fusion) {
   RolesMap roles_map;
 
   // Handle fusion input TensorView objects
-  bool handling_output = false;
   resolveTvToMatmulDomainsMapping(
       deps_map, mma_input_candidates, m, n, k, ca_map);
-  findRolesByDomains(deps_map, roles_map, handling_output);
+  findInputRolesByDomains(deps_map, roles_map);
 
   deps_map.clear();
 
   // Handle fusion output TensorView objects
-  handling_output = true;
   resolveTvToMatmulDomainsMapping(
       deps_map, mma_output_candidates, m, n, k, ca_map);
-  findRolesByDomains(deps_map, roles_map, handling_output);
+  findOutputRolesByDomains(deps_map, roles_map);
+
+  for (auto& [role, tvs] : roles_map) {
+    // sort tvs in descending order by uses() size, and if equal then then by
+    // name() done to make decide which tv should be a reference output tensor
+    // view, if there are more output tvs that match requirements
+    std::sort(tvs.begin(), tvs.end(), [](TensorView* a, TensorView* b) {
+      return (a->uses().size() == b->uses().size())
+          ? (a->name() < b->name())
+          : (a->uses().size() > b->uses().size());
+    });
```

Comment:
Aren't `tvs` already sorted by `findOutputRolesByDomains`?

Reply:
Good catch, I will change it.

The thing is that indeed Fusion output tensors are sorted prior to this loop but inputs are not.

Reply:
I updated the sources:
- for input we will sort all roles (there are only input roles at the time we call `findInputRolesByDomains`)
- for output we will sort tensors during the call to `findOutputRolesByDomains` and assign roles (`OUTPUT_D`, our future reference tensor, and `OUTPUT_AUX`)